### ISSUE TITLE
mix format seems to require :plugins to be set to work

### ISSUE THREAD

**Issue by jeregrine** [elixir-lang/elixir#11550]
### Environment

* Elixir & Erlang/OTP versions (elixir --version): 
```
Erlang/OTP 24 [erts-12.0.3] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1]
Elixir 1.13.1 (compiled with Erlang/OTP 24)
```
* Operating system:  macOS 12.2 Beta (21D5025f), arm64

### Current behavior
When upgrading from elixir v1.12 to v1.13 it our formatter started reporting the following error:

```
❯ mix format
mix format failed for file: test/app_web/views/my_view_test.exs
** (KeyError) key :plugins not found in: [file: "/Users/me/projects/my_project/test/app_web/views/my_view_test.exs", import_deps: [:ecto, :phoenix, :plug, :phoenix_live_view], inputs: ["*.{ex,exs}", "{config,lib,test}/**/*.{ex,exs}"], locals_without_parens: [defguardp: 2, add: 2, add: 3, belongs_to: 2, create: 1, create_if_not_exists: 1, drop: 1, drop_if_exists: 1, field: 2, from: 2, has_many: 2, has_one: 2, modify: 2, modify: 3, remove: 1, rename: 3, defenum: 2, defenum: 3, attributes: 1, location: 1, on_mount: 1, live_dashboard: 1, live_dashboard: 2, oban_dashboard: 1, oban_dashboard: 2, from: 2, field: 1, field: 2, field: 3, timestamps: 1, belongs_to: 2, belongs_to: 3, has_one: 2, has_one: 3, has_many: 2, has_many: 3, many_to_many: 2, many_to_many: 3, embeds_one: 2, embeds_one: 3, embeds_one: 4, embeds_many: 2, embeds_many: 3, embeds_many: 4, intercept: 1, connect: 3, ...]]
    (elixir 1.13.1) lib/keyword.ex:559: Keyword.fetch!/2
    (mix 1.13.1) lib/mix/tasks/format.ex:541: Mix.Tasks.Format.elixir_format/2
    (mix 1.13.1) lib/mix/tasks/format.ex:561: Mix.Tasks.Format.format_file/2
    (elixir 1.13.1) lib/task/supervised.ex:89: Task.Supervised.invoke_mfa/2
    (elixir 1.13.1) lib/task/supervised.ex:34: Task.Supervised.reply/4
    (stdlib 3.15.2) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
```

Here is my `.formatter.exs`
```elixir
[
  import_deps: [:ecto, :phoenix, :plug, :phoenix_live_view],
  inputs: [
    "*.{ex,exs}",
    "{config,lib,test}/**/*.{ex,exs}"
  ],
  locals_without_parens: [
...
  ]
]
```

By applying the following diff it was resolved:

```diff
index 74c015c5f..b48012071 100644
--- a/my_project/.formatter.exs
+++ b/my_project/.formatter.exs
@@ -43,5 +43,6 @@
-  ]
+  ],
+  plugins: []
 ]
```

### Expected behavior

I would expect it to either warn or skip checking for plugins if the key is missing, the line that blew up assumes the plugins key is set and returns.  We could change to Keyword.fetch( and put an else condition on the with... 

BUT I am not sure if I've set something up wrong or if we should change the docs to make plugins required or if we should alter this to warn instead of die.

**Comment by josevalim** [comment]
Can you try running `mix format` on Livebook? It doesn't have the plugins option and it works for me. So I suspect something else is at play here.

**Comment by jeregrine** [comment]
You are right HRM. I tried including the import_deps phoenix/plug/liveview and it still worked fine on livebook but not on my project.

**Comment by jeregrine** [comment]
Gonna chalk this one up to /shrug

**Comment by whatyouhide** [comment]
This actually happened to me too at some point, but I can't remember when or where 😅 Just wanted to voice it out in case other folks come across this 👍

**Comment by jeregrine** [comment]
@whatyouhide did it happen to be an umbrella?

**Comment by whatyouhide** [comment]
@jeregrine nope, that I'm sure of 🙃

**Comment by matthijsqd** [comment]
Just happened to me. After adding the `:plugins` key and running `mix format` once, I could remove it again with no issues. Also, it only occurred for one specific file, and that file had not been edited in 11 months. The error seemed to occur randomly, i had formatted just minutes before that without issue and not touched any configuration.

Not really an issue, but certainly a strange bug.

**Comment by chakeresa** [comment]
This has recently happened to multiple people on our team as well. Adding `plugins: []` to the offending file has fixed it for both of us. Each of us had different files (in the same umbrella) getting mad.

**Comment by josevalim** [comment]
Thanks for the report! I believe this should be fixed on v1.13.3. Please let me know if anyone has seen it on v1.13.3 or later. :)
